licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -63,25 +64,25 @@
  * The encryptedContent is the encrypted representation of the {@link DecryptedLicenseData}:
  * {@code ExpirationDateInMs}
  * {@code issuedTo}
+ * {@code maxNumberOfNodes}
  *
  * For {@link LicenseType#TRIAL} licenses, we use symmetric Cryptography,
  * while for {@link LicenseType#ENTERPRISE} licenses, asymmetric.
  *
- * */
+ */
 public class LicenseService extends AbstractLifecycleComponent implements ClusterStateListener {
 
     private final TransportSetLicenseAction transportSetLicenseAction;
     private final ClusterService clusterService;
-    private final boolean enterpriseEnabled;
 
     private AtomicReference<DecryptedLicenseData> currentLicense = new AtomicReference<>();
+    private AtomicBoolean isMaxNumberOfNodesExceeded = new AtomicBoolean(false);
 
     @Inject
     public LicenseService(Settings settings,
                           TransportSetLicenseAction transportSetLicenseAction,
                           ClusterService clusterService) {
         super(settings);
-        enterpriseEnabled = SharedSettings.ENTERPRISE_LICENSE_SETTING.setting().get(settings);
```

Comment:
Why are we removing this already? For 3.3 the users should still have the option to disable enterprise and not have any licensing restriction. (And later on we need to detect whether enterprise modules are loaded or not)